src/sentry/api/endpoints/project_release_file_details.py

```diff
@@ -35,22 +60,77 @@ def download(releasefile):
         )
         return response
 
-    @classmethod
-    def get_releasefile(cls, request, release, file_id, check_permission_fn):
+    @staticmethod
+    def download_from_archive(release, entry):
+        archive_ident = entry["archive_ident"]
+
+        # Do not use ReleaseFileCache here, we view download as a singular event
+        archive_file = ReleaseFile.objects.get(release=release, ident=archive_ident)
+        archive = ZipFile(archive_file.file.getfile())
+        fp = archive.open(entry["filename"])
+        headers = entry.get("headers", {})
+
+        response = StreamingHttpResponse(
```

Comment:
Why not `HttpResponse(fp)`? This would auto-close afaict

Reply:
I went with [FileResponse](https://docs.djangoproject.com/en/1.11/ref/request-response/#fileresponse-objects), which is a subclass of `StreamingHttpResponse`.